src/Core/Scheduler/Providers/TileProvider.js

```diff
@@ -49,47 +42,78 @@ TileProvider.prototype.preprocessDataLayer = function preprocessLayer(layer, vie
     });
 };
 
+const worldQuaternion = new THREE.Quaternion();
 TileProvider.prototype.executeCommand = function executeCommand(command) {
-    var extent = command.extent;
+    const extent = command.extent;
     if (command.requester &&
         !command.requester.material) {
         // request has been deleted
         return Promise.reject(new CancelledCommandException(command));
     }
+    const layer = command.layer;
+    const builder = layer.builder;
+    const parent = command.requester;
+    const level = (command.level === undefined) ? (parent.level + 1) : command.level;
 
-    var parent = command.requester;
+    if (!this.cacheGeometry[level]) {
+        this.cacheGeometry[level] = new Map();
+    }
 
+    const ce = builder.getCommonGeometryExtent(extent);
+    const south = ce.south().toFixed(8);
+
+    let geometry = this.cacheGeometry[level].get(south);
```

Comment:
I already commented about this: you can't use `south` as a cache key - and no it won't wait for another PR because it can break today.
Additional things to used for the key: builder type, segment count and probably disableSkirt.

Also I'd prefer a single cache than 1 cache per level.

Last but not least: when are you cleaning this cache?
If I zoom in a lot and zoom out, the geometries for high levels will never get disposed..